src/hotspot/share/opto/callGenerator.cpp

```diff
@@ -1018,22 +1018,28 @@ CallGenerator* CallGenerator::for_method_handle_inline(JVMState* jvms, ciMethod*
       Node* receiver = kit.argument(0);
       if (receiver->Opcode() == Op_ConP) {
         input_not_const = false;
-        const TypeOopPtr* oop_ptr = receiver->bottom_type()->is_oopptr();
-        ciMethod* target = oop_ptr->const_oop()->as_method_handle()->get_vmtarget();
-        const int vtable_index = Method::invalid_vtable_index;
+        ciObject* recv_obj = receiver->bottom_type()->is_oopptr()->const_oop();
```

Comment:
In general ConP could be `NULL_PTR` (AnyPtr). I think you need to use `isa_oopptr()` and check for `NULL` here.
Or it can't be `NULL_PTR` in this case?

Reply:
Good catch, Vladimir. I fixed the code and enhanced the test to exercise problematic case by C2.